src/metabase/models/secret.clj

```diff
@@ -100,6 +101,24 @@
           (.write out v)))
       tmp-file)))
 
+(def
+  ^java.io.File
+  ^{:arglists '([{:keys [connection-property-name id value] :as secret} driver?])}
+  value->file!
+  "Returns the value of the given `secret` instance in the form of a file. If the given instance has a `:file-path` as
+  its source, a `File` referring to that is returned. Otherwise, the `:value` is written to a temporary file, which is
+  then returned.
+
+  `driver?` is an optional argument that is only used if an ostensibly existing file value (i.e. `:file-path`) can't be
+  resolved, in order to render a more user-friendly error message (by looking up the display names of the connection
+  properties involved)."
+  (memoize/memo
```

Comment:
It came to my mind (while sleeping), that this changes ever so slightly an invariant: before we would "guarantee" that the file exists after the call. So before we had the ability to recover from lost files without restart.

This is not big issue probably, but I think it's good to be aware of.

Reply:
That's a good point. I'm looking at how it works:

For file paths, it either exists or it throws an error. We lose the error checking each time it is used but that seems not a huge change.

 For uploaded files, it creates a temp file itself. I think its reasonable for the application to assume that file will exist for the lifetime of the program. If someone nukes the temp directory that seems outside of things we should handle. 

So I think this is fine but that's a great edge case to think about.